Update to support mqtt based triggers

yarafie opened the pull request:
- Update to use MQTT based triggers.

Thank you for taking the time to work on a Pull Request. Your contribution is really appreciated! :tada:
**Please don't delete any part of the template**, since keeping the provided structure will help maintainers to review your work more rapidly.

Sections marked as \* are required and need to be filled in.

## Breaking change

This may break existing automations based on the older version for IKEA Tradfri E1743 if being used.
A redownload/update/resetup of the ikea_1743.yaml will probably be needed.

## Proposed change\*

Updated IKEA TRÅDFRI On/Off Switch & Dimmer (E1743) controller to additionally use mqtt triggers. 
Original entity sensor.\<DEVICE\>_action triggers will still work if that is how your system is setup.

Added new controller for IKEA RODRET Dimmer (E2202).

Updated light, cover and media_player hooks to add support for IKEA RODRET Dimmer (E2202).

The new 2 controllers should now work with future releases of zigbee2mqtt 2.0.0 if and when triggers are refactored to use mqtt rather than the soon to be depecrated sensor.\<DEVICE\>_action.

## Checklist\*

- [Y] I followed sections of the [Contribution Guidelines](https://github.com/lsismeiro/awesome-ha-blueprints/blob/main/CONTRIBUTING.md) relevant to changes I'm proposing.
- [Y] I properly tested proposed changes on my system and confirm that they are working as expected.
- [Y] I formatted files with Prettier using the command `npm run format` before submitting my Pull Request.

yarafie committed:
Updated IKEA E1743 TRÅDFRI On/Off Switch & Dimmer controller to use mqtt triggers.
The controller should now work with the upcomming release of zigbee2mqtt 2.0.0 as triggers were refactored to use mqtt rather than the depecrated sensor.<DEVICE>_action.

yarafie committed:
Updated for zigbee2mqtt v2.0.0 mqtt triggers

yarafie committed:
Fixed misplaced commments

yarafie committed:
Optimized code

yarafie committed:
Updated for zigbee2mqtt v2.0.0 mqtt triggers

yarafie committed:
Updated for Zigbee2MQTT_V2

yarafie committed:
Updated for Zigbee2MQTT_V2

LewisSpring commented:
Hiya!
Latest version of this commit looking really good, thanks :)!

Tested my E2001/E2002 and appears to be working perfectly so far!

Edit: That's alone and with the Light Hook 👍

yarafie commented:
Good to know my code actually works :)
Ive already worked on all IKEA Controllers and the sonoff one, they are ready to be tested.
Not sure how to make another pull request though.

I think my changes are pretty easy to follow. If yiu want do one of the phillips or xiomi  and I can check it for you.

LewisSpring commented:
If you need to make any changes, click here 
![image](https://github.com/user-attachments/assets/1a822de2-b0bc-4ce4-a2fa-3ae58fa6199e)
then you can use the GitHub web editor (or git) to upload your changes.

You can keep committing to this same pull request by editing your fork/branch :)

I'll have a look at changing one now.

LewisSpring commented:
For anyone else looking to commit: I would say you should access @yarafie 's repo as per my note above and commit a change to keep this within the same pull too.

LewisSpring committed:
Update ikea_e2001_e2002.yaml

Remove "reported to work with" as Rodret has been moved out to it's own blueprint now.

yarafie commented:
> If you need to make any changes, click here ![image](https://private-user-images.githubusercontent.com/40576482/397976941-1a822de2-b0bc-4ce4-a2fa-3ae58fa6199e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4NjE2NzQsIm5iZiI6MTczNDg2MTM3NCwicGF0aCI6Ii80MDU3NjQ4Mi8zOTc5NzY5NDEtMWE4MjJkZTItYjBiYy00Y2U0LWEyZmEtM2FlNThmYTYxOTllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDA5NTYxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkZmE3ZTMyY2U5Yzg2NTEyZDgwYzllY2Y5MDgzYWYwYTcxN2JlOTFmYWE2MGE4N2E1YzlmMDRlMzAwYTA5ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._U9ENlgRsrw7zYTfnXanlxxfvTKHKYT-0rpgyDww4zk) then you can use the GitHub web editor (or git) to upload your changes.
> 
> You can keep committing to this same pull request by editing your fork/branch :)
> 
> I'll have a look at changing one now.

Already updated my fork yesterday.

LewisSpring commented:
> > If you need to make any changes, click here ![image](https://private-user-images.githubusercontent.com/40576482/397976941-1a822de2-b0bc-4ce4-a2fa-3ae58fa6199e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4NjE2NzQsIm5iZiI6MTczNDg2MTM3NCwicGF0aCI6Ii80MDU3NjQ4Mi8zOTc5NzY5NDEtMWE4MjJkZTItYjBiYy00Y2U0LWEyZmEtM2FlNThmYTYxOTllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDA5NTYxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkZmE3ZTMyY2U5Yzg2NTEyZDgwYzllY2Y5MDgzYWYwYTcxN2JlOTFmYWE2MGE4N2E1YzlmMDRlMzAwYTA5ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._U9ENlgRsrw7zYTfnXanlxxfvTKHKYT-0rpgyDww4zk) then you can use the GitHub web editor (or git) to upload your changes.
> > You can keep committing to this same pull request by editing your fork/branch :)
> > I'll have a look at changing one now.
> 
> Already updated my fork yesterday.

Yep of course! Once all the needed changes are made and tested if possible (though as you have architected this as non-breaking I am not too worried) I will merge it all at once.

yarafie committed:
Merge branch 'main' of github.com:yarafie/awesome-ha-blueprints

yarafie commented:
Looks like all my changes made it as part of the latest PR.
Ive done all the IKEA controllers and the 1 sonoff .
Phillips and xiomai controllers need to probably be done.

Check my code changes (use the ikea_e1743.yaml) as the base.
You can easily follow the changes ive done.
Try and do one of the phillips or xiomai and I can review it if you want.
Good to have at least someone else know.

yarafie committed:
Fixed zha selector

yarafie committed:
Added manufacturer and model for deconz selector.

yarafie committed:
Added manufacturer and model for deconz selector

yarafie committed:
Added Manufacturer and Model for input selectors

yarafie committed:
Added Manufacturer and model to selector

yarafie committed:
Added manufacturer and model to selector

yarafie committed:
Added manufacturer and model to selector

yarafie committed:
Added manufacturer and model to selector

LewisSpring commented:
> Looks like all my changes made it as part of the latest PR. Ive done all the IKEA controllers and the 1 sonoff . Phillips and xiomai controllers need to probably be done.
> 
> Check my code changes (use the ikea_e1743.yaml) as the base. You can easily follow the changes ive done. Try and do one of the phillips or xiomai and I can review it if you want. Good to have at least someone else know.

I had started looking at a Phillips controller this morning - I'll try get one done today.

malgoe commented:
Hey @yarafie 
The blueprint for E1744 has an indentation problem, one space too much on rows 220-250. I just edited it locally and other than that it works great for z2m v2.

yarafie committed:
Fixed indentation in ikea_e1744.yaml
Changed Versiom to 2025-01-01 Happy New Year

yarafie commented:
> Hey @yarafie The blueprint for E1744 has an indentation problem, one space too much on rows 220-250. I just edited it locally and other than that it works great for z2m v2.

Thanks. Fixed it.

LewisSpring committed:
Merge branch 'main' into main

LewisSpring commented:
Updated [ikea_e1744.yaml](https://github.com/lsismeiro/awesome-ha-blueprints/pull/33/commits/104bf9617252372077e924adb52192752ff19037#diff-a3ef010dce3eb82b9c7f237a2705daadf8c6a75b732e74c1e88e3abb1fc0bea0) version to tomorrows date to resolve merge conflict. 

This also makes it clear that the fix from #29 is applied - I know it was there already :)

LewisSpring committed:
Update philips_324131092621.yaml

Copied new elements from e2001/e2002 excluding the device selector.

LewisSpring commented:
@yarafie if you can review https://github.com/lsismeiro/awesome-ha-blueprints/pull/33/commits/6ef7783d453b563634d4bc34d5c2e5cd98f831d2
that would be great!

Apologies for the delay in doing this, both the right and wrong time of year :D

LewisSpring commented:
> @yarafie if you can review [6ef7783](https://github.com/lsismeiro/awesome-ha-blueprints/commit/6ef7783d453b563634d4bc34d5c2e5cd98f831d2) that would be great!
> 
> Apologies for the delay in doing this, both the right and wrong time of year :D

Of course with thanks for getting started on this - I wouldn't clue where to start without doing a diff of the old and new e2001 blueprint.

I will probably try to finish the next few tonight for review, we can merge hopefully before Z2M v2 drops.

LewisSpring committed:
Update remaining Osram, Philips and Xiaomi blueprints. Phew!

LewisSpring commented:
Hi - that's the rest done! Was a bit of a marathon but not too hard to do.

philips_8718699693985.yaml does not support Z2M already so have left for now.
Might be trivial to add just like the others but not 100% sure.

yarafie commented:
@lsismeiro Looks like EPmatt is back after Long hiatus.

Nicolai- commented on blueprints/controllers/philips_324131092621/philips_324131092621.yaml:
```diff
@@ -359,6 +373,67 @@ trigger:
     event_type: state_changed
     event_data:
       entity_id: !input controller_entity
+  # triggers for zigbee2mqtt_v2
```
I tried testing the Philips 324131092621 Hue Dimmer switch and this version of the blueprint.
The subtypes are all wrong :) I think you switched the values from the mapping.
Ex.
`button_on_short` => `on_press`
`button_on_long` => 'on_hold'
`button_on_release` => `on_hold_release`

Nicolai- commented on blueprints/controllers/philips_324131092621/philips_324131092621.yaml:
```diff
@@ -384,6 +461,8 @@ condition:
       # only for zigbee2mqtt, check if the event is relative to a real state change, and not only some minor changes in the sensor attributes
       # this is required since multiple state_changed events are fired for a single button press, with the result of the automation being triggered multiple times
       - '{{ integration_id != "zigbee2mqtt" or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
+      - '{{ integration_id != "zigbee2mqtt_v2" or trigger.payload != last_controller_event }}'
```
`last_controller_event` is never defined as a variable, so the value is always "undefined", so this will always be true.

On "long" presses, multiple events with ex "up_hold" will receive, and each will start a run of the automation.

It seems that the `last_controller_event` is implemented in some of the controllers.
Ex. `ikea_e1743.yaml` but not here

Nicolai- reviewed (commented):
I tried testing the Philips 324131092621 Hue Dimmer switch and this version of the blueprint and it does not work with the v2

See my comments below.

Iv'e been following this on the side line as I also have a fork of the original repo which I adjusted for my own use.

yarafie commented:
@lsismeiro @Nicolai- I'll take a look at it later

yarafie committed:
Added last_controller_rvent checks where applicable

yarafie commented:
@Nicolai- Can you test the latest just did an update

LewisSpring commented:
> @lsismeiro Looks like EPmatt is back after Long hiatus.

I've seen - I did message @lsismeiro on HA forum about next steps for that as _it is his repo after all ;)_ - I will likely send him an email.
For now my intent is to keep this running and I'll merge any changes from upstream if possible.

LewisSpring commented:
> @Nicolai- Can you test the latest just did an update

Thanks for fixing.
Are the mappings still wrong?

LewisSpring commented:
@yarafie How far off are we from merging would you say?

hvorragend commented:
I have seen that there are still some minor problems with the code indentation in yarafie's repo.

LewisSpring commented:
@hvorragend Only just spotted your comments in my email inbox. They did not jump at me on Github itself.
Will look now.

LewisSpring committed:
Fix Xiaomi indentation. Thanks @hvorragend

LewisSpring committed:
Merge branch 'main' into main

LewisSpring commented:
No changes to https://github.com/lsismeiro/awesome-ha-blueprints/pull/33/commits/d1d35b83dbb0cd0107b07ef0a6204a23b1605c28 made. Just fixing conflict with latest merge #35 
Only needed to put the version number back.

yarafie commented:
> I have seen that there are still some minor problems with the code indentation in yarafie's repo. 

I beleive i fixed the code indentation problem, it was in one blueprint.

yarafie commented:
> @yarafie How far off are we from merging would you say?

I think we are ready, maybe some cleanup and consistancy in the blueprints is needed.

alchemistake commented on blueprints/controllers/ikea_e1766/ikea_e1766.yaml:
```diff
@@ -211,6 +252,8 @@ action:
         {%- if integration_id == "zigbee2mqtt" -%}
         {{ trigger.event.data.new_state.state }}
+        {%- elif integration_id == "zigbee2mqtt_v2" -%}
+        {{ trigger.event.data.new_state.state }}
```
This should be
```suggestion
        {{ trigger.payload }}
```
tested it on my box

yarafie replied:
> This should be
> 
> tested it on my box

Oops. Will fix it. Good Catch

theincogtion commented:
@LewisSpring 
https://github.com/lsismeiro/awesome-ha-blueprints/blob/main/blueprints/controllers/ikea_e2001_e2002/ikea_e2001_e2002.yaml
I tried to use this one with z2m 2.0 and legacy disabled. But I am unable to select the correct sensor in the selection dialog:
![image](https://github.com/user-attachments/assets/5fb8ed26-f054-42d9-87d0-9ad0f1a12ec8)
It only shows me the battery sensor.

yarafie committed:
Fixed trigger in ikea_e1766.yaml
Cleaned up code consistency

huggenknubbel commented:
i tried the new Blueprint from here: https://github.com/lsismeiro/awesome-ha-blueprints/blob/aa3aff35b9c80de76ed448d3a2aaae134799e23b/blueprints/controllers/ikea_e2001_e2002/ikea_e2001_e2002.yaml
But i cant select the MQTT device. It said "no match"
![grafik](https://github.com/user-attachments/assets/f53c7504-bf60-41aa-a55f-7855d192e3aa)
i have two of this remotes and checked the device info in mqtt:
![grafik](https://github.com/user-attachments/assets/b3dff353-7774-418d-a863-6a562e2da285)
but it seems like its all fine?

yarafie commented:
> i tried the new Blueprint from here: https://github.com/lsismeiro/awesome-ha-blueprints/blob/aa3aff35b9c80de76ed448d3a2aaae134799e23b/blueprints/controllers/ikea_e2001_e2002/ikea_e2001_e2002.yaml But i cant select the MQTT device. It said "no match" ![grafik](https://private-user-images.githubusercontent.com/15655280/400204654-f53c7504-bf60-41aa-a55f-7855d192e3aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwODgwNzksIm5iZiI6MTczNjA4Nzc3OSwicGF0aCI6Ii8xNTY1NTI4MC80MDAyMDQ2NTQtZjUzYzc1MDQtYmY2MC00MWFhLWE1NWYtNzg1NWQxOTJlM2FhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDE0MzYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwYTllMmRkODY2NWFkNDVjNzVjODA4OTBhMjBjNWIyMTBlMGJkOTEzNmRkM2YwN2Q5OWFhYmNkYTM1ZGUyMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.L9jV2uMfjdEJLRte5rfXW_GUOkKTxs7z0AgQc4HHuR4) i have two of this remotes and checked the device info in mqtt: ![grafik](https://private-user-images.githubusercontent.com/15655280/400204832-b3dff353-7774-418d-a863-6a562e2da285.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwODgwNzksIm5iZiI6MTczNjA4Nzc3OSwicGF0aCI6Ii8xNTY1NTI4MC80MDAyMDQ4MzItYjNkZmYzNTMtNzc3NC00MThkLWE4NjMtNmE1NjJlMmRhMjg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDE0MzYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhMzFkYmY0ZWEyZmVmOGJlZDJkN2M4NzRhYjQ4ZWMyMDliZDEwNmEwYTU5YzQxMWEwZGQ4ODVmNTJmOWYxZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mYc69uoHV7clEB6__Re6uiBHXiNCTkhQqMjJnhN7Aqw) but it seems like its all fine?

Has the STYRBAR Ever worked for you in zigbee2mqtt?
Did you recently upgrade your STYRBAR firmware to 2.4.16?

malgoe commented:
I have a working STYRBAR using this blueprint. Firmware is 2.4.16. Worked in 1.42 with legacy disabled and works fine after 2.0 update.

Maybe I dont have the same version of the blueprint? I got it from @yarafie repo about a week ago?
![Skärmbild från 2025-01-05 16-50-57](https://github.com/user-attachments/assets/b45fbf2c-e637-4852-8905-89235b56e0a1)
![Skärmbild från 2025-01-05 16-50-11](https://github.com/user-attachments/assets/42e20134-9e23-4580-8e97-2530e1cd10a7)

huggenknubbel commented:
> > i tried the new Blueprint from here: https://github.com/lsismeiro/awesome-ha-blueprints/blob/aa3aff35b9c80de76ed448d3a2aaae134799e23b/blueprints/controllers/ikea_e2001_e2002/ikea_e2001_e2002.yaml But i cant select the MQTT device. It said "no match" ![grafik](https://private-user-images.githubusercontent.com/15655280/400204654-f53c7504-bf60-41aa-a55f-7855d192e3aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwODgwNzksIm5iZiI6MTczNjA4Nzc3OSwicGF0aCI6Ii8xNTY1NTI4MC80MDAyMDQ2NTQtZjUzYzc1MDQtYmY2MC00MWFhLWE1NWYtNzg1NWQxOTJlM2FhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDE0MzYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwYTllMmRkODY2NWFkNDVjNzVjODA4OTBhMjBjNWIyMTBlMGJkOTEzNmRkM2YwN2Q5OWFhYmNkYTM1ZGUyMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.L9jV2uMfjdEJLRte5rfXW_GUOkKTxs7z0AgQc4HHuR4) i have two of this remotes and checked the device info in mqtt: ![grafik](https://private-user-images.githubusercontent.com/15655280/400204832-b3dff353-7774-418d-a863-6a562e2da285.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwODgwNzksIm5iZiI6MTczNjA4Nzc3OSwicGF0aCI6Ii8xNTY1NTI4MC80MDAyMDQ4MzItYjNkZmYzNTMtNzc3NC00MThkLWE4NjMtNmE1NjJlMmRhMjg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDE0MzYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhMzFkYmY0ZWEyZmVmOGJlZDJkN2M4NzRhYjQ4ZWMyMDliZDEwNmEwYTU5YzQxMWEwZGQ4ODVmNTJmOWYxZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mYc69uoHV7clEB6__Re6uiBHXiNCTkhQqMjJnhN7Aqw) but it seems like its all fine?
> 
> Has the STYRBAR Ever worked for you in zigbee2mqtt? Did you recently upgrade your STYRBAR firmware to 2.4.16?

yes. the integration worked until i upgrated to Zigbee2MQTT V2. if i enable the Legacy option in Z2MQTT, the remote worked again and i have an other Blueprint where the Remote is working without legacy options. The only difference between this Blueprint here, and the working one at the device selector is the other one has no filter. But if i delete the filter in Blueprint i still can't chose any device.  

Edit: 
after a HA Restart i can find the IKEA Remote. But only if i delete the filter out of the Blueprint

huggenknubbel commented:
OK i found the bug. The filter has to be:
```
         - integration: mqtt
            manufacturer: IKEA
            model: STYRBAR remote control
```

malgoe commented:
> OK i found the bug. The filter has to be:
> 
> ```
>          - integration: mqtt
>             manufacturer: IKEA
>             model: STYRBAR remote control
> ```

Will that cover my device as well? "Remote Control N2". Though the description field says "STYRBAR remote control" on mine. (see my screenshot in previous comment)

yarafie commented:
> OK i found the bug. The filter has to be:
> 
> ```
>          - integration: mqtt
>             manufacturer: IKEA
>             model: STYRBAR remote control
> ```

Which z2m version u r using?

tivoo commented:
Because I got the same No compatible device found error, I tried to get rid off the filter completely but I still seem to get the error. I tried to replicate the same code asVersion 2023.10.05 from @lsismeiro . But unfortunately it doesn't work:

```    
controller_device:
      name: (deCONZ, ZHA, Zigbee2MQTT_V2) Controller Device
      description: The controller device to use for the automation. Choose a value
        only if the remote is integrated with deCONZ, ZHA, Zigbee2MQTT_V2.
      default: ''
      selector:
        device: {}
```

Am I missing something? Would love to test out the blueprints.

zigbee2mqtt version 2.0.0. 

Device info:
Zigbee model Remote Control N2
Zigbee Fabrikant IKEA of Sweden
Omschrijving STYRBAR remote control
Ondersteuningsstatus Ondersteund
IEEE adres 0xcc86ecfffe6aa825
Netwerkadres 0xB2E2 / 45794
Firmware build date 20240708
Firmwareversie 2.4.16
Fabrikant [IKEA](https://www.zigbee2mqtt.io/supported-devices/#v=IKEA)
Model [E2001/E2002](https://www.zigbee2mqtt.io/devices/E2001_E2002.html#ikea-e2001%252Fe2002)

yarafie commented:
> OK i found the bug. The filter has to be:
> 
> ```
>          - integration: mqtt
>             manufacturer: IKEA
>             model: STYRBAR remote control
> ```

Can you in Home Assistant click on the MQTT INFO for your device and share with me the expanded first payload would like to see what you have.

Mine is attached

![Screenshot_20250105_184509_Home Assistant](https://github.com/user-attachments/assets/68d185fc-8888-4a3e-8823-db8a79481d4c)

Nicolai- commented:
> > @Nicolai- Can you test the latest just did an update
> 
> Thanks for fixing. Are the mappings still wrong?

@yarafie 
They still don't work, as soon as I change the integration type to the v2, it does work.

If I change the "subtype" of the triggers to the value from the "mapping array", then it works.

If you are trying to use the `button_x` variables from the `variables` section defined from the mapping, it won't work. As these variables cannot be used in the `trigger` section to my knowledge.

There is also now an missing space on line 650 after the `-`

huggenknubbel commented:
> > OK i found the bug. The filter has to be:
> > ```
> >          - integration: mqtt
> >             manufacturer: IKEA
> >             model: STYRBAR remote control
> > ```
> 
> Can you in Home Assistant click on the MQTT INFO for your device and share with me the expanded first payload would like to see what you have.
> 
> Mine is attached
> 
> ![Screenshot_20250105_184509_Home Assistant](https://private-user-images.githubusercontent.com/88841013/400211585-68d185fc-8888-4a3e-8823-db8a79481d4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwOTIzMDMsIm5iZiI6MTczNjA5MjAwMywicGF0aCI6Ii84ODg0MTAxMy80MDAyMTE1ODUtNjhkMTg1ZmMtODg4OC00YTNlLTg4MjMtZGI4YTc5NDgxZDRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDE1NDY0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkYWQ0ZTljNGY2MGYzZGFiYTZkNDJlM2VmN2JkOTZiOTY2YzY5YWM5NGY5YTA0ZjNmNWQzNzM5ZWQ5YWE3MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5hld2LQMLQbyY4r2gwa6qQWk8c-KZuUoJpGxmefZUkc)

you talking about this? 
![grafik](https://github.com/user-attachments/assets/3521f99e-4b34-436f-b5f6-7e0061356e99)

yarafie commented:
> > > OK i found the bug. The filter has to be:
> > > ```
> > >          - integration: mqtt
> > >             manufacturer: IKEA
> > >             model: STYRBAR remote control
> > > ```
> > 
> > 
> > Can you in Home Assistant click on the MQTT INFO for your device and share with me the expanded first payload would like to see what you have.
> > Mine is attached
> > ![Screenshot_20250105_184509_Home Assistant](https://private-user-images.githubusercontent.com/88841013/400211585-68d185fc-8888-4a3e-8823-db8a79481d4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwOTIzMDMsIm5iZiI6MTczNjA5MjAwMywicGF0aCI6Ii84ODg0MTAxMy80MDAyMTE1ODUtNjhkMTg1ZmMtODg4OC00YTNlLTg4MjMtZGI4YTc5NDgxZDRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDE1NDY0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkYWQ0ZTljNGY2MGYzZGFiYTZkNDJlM2VmN2JkOTZiOTY2YzY5YWM5NGY5YTA0ZjNmNWQzNzM5ZWQ5YWE3MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5hld2LQMLQbyY4r2gwa6qQWk8c-KZuUoJpGxmefZUkc)
> 
> you talking about this? ![grafik](https://private-user-images.githubusercontent.com/15655280/400212582-3521f99e-4b34-436f-b5f6-7e0061356e99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwOTM5NzAsIm5iZiI6MTczNjA5MzY3MCwicGF0aCI6Ii8xNTY1NTI4MC80MDAyMTI1ODItMzUyMWY5OWUtNGIzNC00MzZmLWI1ZjYtN2UwMDYxMzU2ZTk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDE2MTQzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3OTM1NTM4YTVkOWUzNjQ0MzJiM2Y3ZDA3MmM3Y2QyNTBlMjUyMmE3YTI2MGVmZDdiMjljYTE5OWExZDRjN2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-FWNIC9BkLmJSz0y56maZADl3-EEBubht3KIlLfUKP0)

I think z2m 2.0.0 changed the naming convemtion. I'm still using z2m 1.42 and you can see some differences. I don't have model_id as seperate info for me its part of model. That is why its not finding it for you in selector but works for me.

I need to rethink the selector search :)

yarafie committed:
modified:   philips_324131092621.yaml Fixed triggers subtype for zigbee2mqtt v2

yarafie commented:
> > > @Nicolai- Can you test the latest just did an update
> > 
> > 
> > Thanks for fixing. Are the mappings still wrong?
> 
> @yarafie They still don't work, as soon as I change the integration type to the v2, it does work.
> 
> If I change the "subtype" of the triggers to the value from the "mapping array", then it works.
> 
> If you are trying to use the `button_x` variables from the `variables` section defined from the mapping, it won't work. As these variables cannot be used in the `trigger` section to my knowledge.
> 
> There is also now an missing space on line 650 after the `-`

I fixed it. The triggers were wrong as you mentioned.

EPMatt commented on blueprints/controllers/ikea_e1524_e1810/ikea_e1524_e1810.yaml:
```diff
@@ -1,11 +1,14 @@
 blueprint:
+  domain: automation
+  homeassistant:
+    min_version: 2023.6.0
```
Question - why is this HA Version requirement needed?

LewisSpring replied:
I believe this was to do with the trigger_delta: fixes implemented in the early life of redux - I think this is now redundant by this commit. 

I am confident that this can be removed across the blueprints.

LewisSpring replied:
Removed with https://github.com/lsismeiro/awesome-ha-blueprints/pull/33/commits/e399f691317eb729ad3d52fb1471f78af284c305

EPMatt commented on blueprints/controllers/ikea_e1524_e1810/ikea_e1524_e1810.yaml:
```diff
@@ -31,12 +31,24 @@ blueprint:
             - deCONZ
             - ZHA
             - Zigbee2MQTT
+            - Zigbee2MQTT_V2
```
```suggestion
```

EPMatt commented on blueprints/controllers/ikea_e1524_e1810/ikea_e1524_e1810.yaml:
```diff
@@ -31,12 +31,24 @@ blueprint:
     controller_device:
-      name: (deCONZ, ZHA) Controller Device
-      description: The controller device to use for the automation. Choose a value only if the remote is integrated with deCONZ, ZHA.
+      name: (deCONZ, ZHA, Zigbee2MQTT_V2) Controller Device
+      description: The controller device to use for the automation. Choose a value only if the remote is integrated with deCONZ, ZHA, Zigbee2MQTT_V2.
```
```suggestion
      name: (deCONZ, ZHA, Zigbee2MQTT) Controller Device
      description: The controller device to use for the automation. Choose a value only if the remote is integrated with deCONZ, ZHA, Zigbee2MQTT.
```

Instead of considering Z2M v2.0.0 as a separate integration, I'd consider it as the actual "Zigbee2MQTT" integration without distinctions. To check whether legacy triggers want to be used, we can simply check whether the user provided a `controller_entity`.

To me it sounds like this approach would make it easier to drop support for legacy triggers in the future - just remove the `controller_entity` input + any business logic related to it.

LewisSpring replied:
I think that is a more user-friendly way of doing it and aides in tech debt for future versions.

@yarafie would you know where to go with this suggestion, if you would like to take it up?
Would @EPMatt be able to support if needed on this?

Nicolai- replied:
Just some observations.
I tried rewriting my philips controller to this (as that I could test and agree that it's more user friendly).

The new triggers they require a valid device id, which is only the case if a Controller Device is selected.
So when the blueprint is updated the automations becomes unavailable until a controller device has been selected.

So I don't think it's possible to avoid a breaking change unless choosing another route.
I added my changes here [https://github.com/Nicolai-/awesome-ha-blueprints/tree/ns_mqtt_device_triggers](https://github.com/Nicolai-/awesome-ha-blueprints/tree/ns_mqtt_device_triggers)

There is another way to integrate and that is the MQTT Trigger which can listen on a [mqtt topic](https://www.home-assistant.io/docs/automation/trigger/#mqtt-trigger).
The main drawback from that is that the topic should be entered manually by the user
Atleast I haven't found a way to fetch it from a device/entity.

EPMatt commented on blueprints/controllers/ikea_e1524_e1810/ikea_e1524_e1810.yaml:
```diff
@@ -412,6 +510,8 @@ condition:
         {%- if integration_id == "zigbee2mqtt" -%}
         {{ trigger.event.data.new_state.state }}
+        {%- elif integration_id == "zigbee2mqtt_v2" -%}
+        {{ trigger.payload }}
```
Same here (and in any other spot where we check for `integration_id == "zigbee2mqtt"`) to check whether legacy triggers want to be used, we can simply check whether integration is `zigbee2mqtt` and the user provided a `controller_entity`.

LewisSpring replied:
As [here](https://github.com/lsismeiro/awesome-ha-blueprints/pull/33/files/98c6b02d21dabdbb42cb058f18aec33b0637b4a1#r1903313178)

EPMatt commented on blueprints/hooks/cover/cover.yaml:
Just as a side note, since this PR is starting to grow considerably, I'd keep extra unrelated changes to hooks + new controllers to separate PRs. 👍🏻

LewisSpring replied:
Same note as [E2201 ](https://github.com/lsismeiro/awesome-ha-blueprints/pull/33/files/98c6b02d21dabdbb42cb058f18aec33b0637b4a1#r1903316112)

EPMatt commented on blueprints/hooks/light/light.yaml:
Same here, Just as a side note, since this PR is starting to grow considerably, I'd keep extra unrelated changes to hooks + new controllers to separate PRs. 👍🏻

LewisSpring replied:
Same note as [E2201 ](https://github.com/lsismeiro/awesome-ha-blueprints/pull/33/files/98c6b02d21dabdbb42cb058f18aec33b0637b4a1#r1903316112)

EPMatt commented on blueprints/hooks/media_player/media_player.yaml:
Same here, Just as a side note, since this PR is starting to grow considerably, I'd keep extra unrelated changes to hooks + new controllers to separate PRs. 👍🏻

LewisSpring replied:
Same note as [E2201 ](https://github.com/lsismeiro/awesome-ha-blueprints/pull/33/files/98c6b02d21dabdbb42cb058f18aec33b0637b4a1#r1903316112)

EPMatt commented on blueprints/controllers/ikea_e2201/ikea_e2201.yaml:
Same here, Just as a side note, since this PR is starting to grow considerably, I'd keep extra unrelated changes to hooks + new controllers to separate PRs. 👍🏻

LewisSpring replied:
@yarafie Could we move this to a branch on your repo instead of main to resolve this?

EPMatt reviewed (changes requested):
Hey there, EPMatt from the original repo here 👋🏻  

First of all, I want to thank everyone who's supported this project while I was away - life happens, and unfortunately I couldn't contribute at all to this project for a long time. Luckily, I'm hopefully now slowly getting back to be more active in the [original repo](https://github.com/EPMatt/awesome-ha-blueprints).

I see the fix for Z2M 2.0.0 is going strong here! 🔥  I'm leaving here a few comments to provide better retro-compatibility and ease phase out of legacy triggers at a later stage. I left suggestions on only one controller file, but consider them replicated to all the controllers involved in this contribution.

This review is open to feedback. Let me know what you think!

waclaw66 commented:
Seems double press helper doesn't work at least for RODRET (E2201). Triggers for ON and OFF are saved into _Last Controller Event_ always as double_press.
`{"a":"double_press","t":1736102343.649046}`
_Double Press delay_ set to 500ms

Nicolai- commented:
> > > > @Nicolai- Can you test the latest just did an update
> > > 
> > > 
> > > Thanks for fixing. Are the mappings still wrong?
> > 
> > 
> > @yarafie They still don't work, as soon as I change the integration type to the v2, it does work.
> > If I change the "subtype" of the triggers to the value from the "mapping array", then it works.
> > If you are trying to use the `button_x` variables from the `variables` section defined from the mapping, it won't work. As these variables cannot be used in the `trigger` section to my knowledge.
> > There is also now an missing space on line 650 after the `-`
> 
> I fixed it. The triggers were wrong as you mentioned.

It works now :)

EPMatt commented:
I want to truly thank @lsismeiro , @LewisSpring , @yarafie and everyone else who contributed to this fork! You helped to keep this project alive, and it's amazing and super motivating to see how much it grew while I was away! 🔥 

I'd be really happy to join efforts and gradually bring the development back to the main repo. Considering the time you all dedicated here, I'd love to discuss and hear thoughts from @lsismeiro and any other maintainer before merging Awesome HA Blueprints - Redux to the original repo.

However, given the urgency of implementing Z2M device triggers, @yarafie I'd suggest opening a separate PR to the original repo replicating the changes discussed here, once this contribution is ready, so we can provide the fix more quickly via the official channel. What do you think?

I'm taking some time to update the original repo - merging PRs, following up on issues and fixing open bugs there - in preparation for the upcoming merge. Let me know if you need any help to proceed with this fix.

Thanks again!

LewisSpring commented:
Hiya @EPMatt 

I really appreciate the kind words.
I was going to send you an email suggesting the same thing!

I haven't heard from Luis in a while but personally I would be happy to commit back towards your repo and it would be great to discuss this further.

In particular, it would be good to identify any commits to this repo - mainly fixes - that have not already been PR'ed to your own repo to be backported.

Thanks for the comments on the commits as well.
I will reply to the ones that I can and am sure @yarafie will be able to handle those I cannot.

Cheers

theincogtion commented:
> Seems double press helper doesn't work at least for RODRET (E2201). Triggers for ON and OFF are saved into _Last Controller Event_ always as double_press. `{"a":"double_press","t":1736102343.649046}` _Double Press delay_ set to 500ms

https://github.com/damru/ha-blueprints/issues/11
same here

LewisSpring reviewed (changes requested):
Added some replies to EPMatt's feedback

I can look at removing the minimum version but rework of the Z2M logic is out of my depth.

LewisSpring committed:
Remove HA Minimum Version

Unrequired minimum versions removed.

LewisSpring commented:
@EPMatt  Regarding the PR in your repo for the V2 addition - are you looking for a new branch on your repo or a PR from @yarafie 's fork/main branch?

dependabot[bot] committed:
github-actions: bump creyD/prettier_action from 4.2 to 4.3 (#523)

dependabot[bot] committed:
npm-website: bump @docusaurus/core from 2.0.1 to 2.3.1 in /website (#506)

dependabot[bot] committed:
npm-website: bump http-cache-semantics from 4.1.0 to 4.1.1 in /website (#507)

dependabot[bot] committed:
github-actions: bump reviewdog/action-eslint from 1.17 to 1.18 (#525)

dependabot[bot] committed:
npm-website: bump eslint from 8.21.0 to 8.35.0 in /website (#526)

dependabot[bot] committed:
npm-website: bump eslint-config-prettier from 8.5.0 to 8.7.0 in /website (#528)

dependabot[bot] committed:
npm-website: bump loader-utils from 2.0.2 to 2.0.4 in /website (#449)

dependabot[bot] committed:
github-actions: bump s0/git-publish-subdir-action from 2.5.1 to 2.6.0 (#451)

dependabot[bot] committed:
npm-website: bump react-bootstrap-icons from 1.8.4 to 1.10.2 in /website (#454)

dependabot[bot] committed:
github-actions: bump dessant/lock-threads from 3 to 4 (#458)

dependabot[bot] committed:
npm-website: bump body-parser and express in /website (#633)

dependabot[bot] committed:
npm-website: bump prettier from 2.7.1 to 2.8.4 in /website (#513)

dependabot[bot] committed:
npm-website: bump @sideway/formula from 3.0.0 to 3.0.1 in /website (#514)

dependabot[bot] committed:
npm-website: bump cross-spawn from 7.0.3 to 7.0.6 in /website (#626)

dependabot[bot] committed:
github-actions: bump actions/cache from 3.0.6 to 3.2.6 (#524)

dependabot[bot] committed:
npm-website: bump ua-parser-js from 0.7.31 to 0.7.33 in /website (#498)

dependabot[bot] committed:
github-actions: bump actions/stale from 5 to 7 (#473)

dependabot[bot] committed:
npm: bump prettier from 2.7.1 to 2.8.4 (#512)

dependabot[bot] committed:
npm-website: bump json5 in /website (#635)

dependabot[bot] committed:
npm-website: bump webpack from 5.73.0 to 5.97.1 in /website (#637)

dependabot[bot] committed:
npm-website: bump micromatch from 4.0.5 to 4.0.8 in /website (#636)

dependabot[bot] committed:
npm-website: bump nanoid from 3.3.4 to 3.3.8 in /website (#628)

dependabot[bot] committed:
npm-website: bump http-proxy-middleware from 2.0.6 to 2.0.7 in /website (#631)

yarafie commented:
First of all Welcome back @EPMatt 

> I want to truly thank @lsismeiro , @LewisSpring , @yarafie and everyone else who contributed to this fork! You helped to keep this project alive, and it's amazing and super motivating to see how much it grew while I was away! 🔥
> 

It is truely my pleasure. I actually Don't use the blueprints myself, but learned so much from the codes that I felt I could give back. I'm treating this like a jigsaw puzzle, and since I really haven't been coding for a while, felt that this would help keep my mind sharp.

> I'd be really happy to join efforts and gradually bring the development back to the main repo. Considering the time you all dedicated here, I'd love to discuss and hear thoughts from @lsismeiro and any other maintainer before merging Awesome HA Blueprints - Redux to the original repo.
> 

I was going to ask the same regarding the original repo (yours). I think it best to just focus on updating the main Repo rather than lsismerio's fork, otherwise things may get confusing. 

> However, given the urgency of implementing Z2M device triggers, @yarafie I'd suggest opening a separate PR to the original repo replicating the changes discussed here, once this contribution is ready, so we can provide the fix more quickly via the official channel. What do you think?
> 
> I'm taking some time to update the original repo - merging PRs, following up on issues and fixing open bugs there - in preparation for the upcoming merge. Let me know if you need any help to proceed with this fix.
> 
> Thanks again!

What I plan to do is:
1- Start from a clean latest clone of your Repo (EPMatt).
2- Work on a new controller ikea_e2201 (Rodret) since I can actually test with it, also update the hooks for the Rodret. This will serve as my base and I can test your suggesions once I figure out how to do it (BTW: I agree with your suggestions just need to see if it is doable)
3-  In parallel Work on updating ikea_e1743 controller (My e1743 is not working at the moment but that is another story)

LewisSpring commented:
@yarafie - that is really brilliant of you and again I really appreciate your help.

I think that's a good idea. I will look at doing a once-over of the current V2 compatible blueprints for this repo and merge them to have something out for now - but I'd be happy to give you a hand if needed/possible on your new V2 fork. I have E2001 and E1743 I can test with.

Please drop the repo link and PR when you make it :)

theincogtion commented:
Coming back to the double press issue:
https://community.home-assistant.io/t/ikea-rodret-dimmer-somrig-remotes-zha-z2m-control-anything/590081/98?u=usefulvid

EPMatt commented:
Hey @yarafie @LewisSpring,
thank you for your nice words. Love to hear that you're in for bringing back your precious contributions back to the original repo! 🔥 

> @EPMatt Regarding the PR in your repo for the V2 addition ..

@LewisSpring, the plan outlined by @yarafie looks good - better to start with a clean fork for this specific Z2M v2.0 fix. We can later merge back the changes from this fork with a PR (Isismeiro/main -> EPMatt/main) after we are done with the changes discussed here. For now, I feel like restoring the functionality of the controller blueprints should be a priority 👍🏻 


> What I plan to do is:
1- Start from a clean latest clone of your Repo (EPMatt).
2- Work on a new controller ikea_e2201 (Rodret) since I can actually test with it, also update the hooks for the Rodret. This will serve as my base and I can test your suggesions once I figure out how to do it (BTW: I agree with your suggestions just need to see if it is doable)
3- In parallel Work on updating ikea_e1743 controller (My e1743 is not working at the moment but that is another story)

@yarafie looks good! Regarding 2 and 3, I'd suggest to break your contribution in smaller PRs, in a way similar to the following:
1. New Controller: IKEA E2201 with Z2M device triggers support: this allows you to test the changes discussed here on your own hardware. For simplicity and speed, I'd suggest to keep the implementation to Z2M only, so we can merge this faster and work on ZHA and Deconz later.
2. (Z2M) Add device triggers support for controllers: after we have approved and merged 1., we can replicate the device trigger changes to all the other controllers. In this way, approving and merging this second PR would be quick (the changes being just a copy-paste of the approach tested and approved in 1.)

What do you think?

Aside from that, once again thank you for your help and commitment to the project. Let me know if you have any questions / doubts or if there's anything I can help you with. 🔥

EPMatt commented:
> Coming back to the double press issue: https://community.home-assistant.io/t/ikea-rodret-dimmer-somrig-remotes-zha-z2m-control-anything/590081/98?u=usefulvid

Thanks for reporting @theincogtion! Could you please open a new Issue in the [original repo](https://github.com/EPMatt/awesome-ha-blueprints)? We have received reports of double presses not working in the past, so this is probably one of the things we're gonna look into next.

yarafie committed:
Added Support for zigbee2mqtt mqtt device triggers when legacy=false
Preserving backwards compatability with zigbee2mqtt legacy entity action events

yarafie committed:
Added Support for zigbee2mqtt mqtt device triggers when legacy=false
Preserving backwards compatability with zigbee2mqtt legacy entity action events

yarafie commented:
@EPMatt @LewisSpring 
Using the base code in EPMatt main.
I have refactored ikea_e1743.yaml with the suggested changes.
I did the same for the new controller ikea_e2201.yaml which I tested and seems to be working at least for the z2m mqtt triggers. I was able to turn on and off a light bulb and also increase decrease (dimming) by holding either button down.

Someone needs to fully test the ikea_e1743.yaml (zigbee2mqtt both legacy and new mqtt and zha and deconz)

I have left a lot of comments in the code to easily see the changes (off course their is diff) but I'm Old School :)

It is committed to this repository lsismeiro.

Next Steps?

yarafie commented:
> > > > > @Nicolai- Can you test the latest just did an update
> > > > 
> > > > 
> > > > Thanks for fixing. Are the mappings still wrong?
> > > 
> > > 
> > > @yarafie They still don't work, as soon as I change the integration type to the v2, it does work.
> > > If I change the "subtype" of the triggers to the value from the "mapping array", then it works.
> > > If you are trying to use the `button_x` variables from the `variables` section defined from the mapping, it won't work. As these variables cannot be used in the `trigger` section to my knowledge.
> > > There is also now an missing space on line 650 after the `-`
> > 
> > 
> > I fixed it. The triggers were wrong as you mentioned.
> 
> It works now :)

I took a look at your code and drew inspiration from it.
Take a look at ikea_e1743.yaml or ikea_e2201.yaml and see what I did their regarding z2m new and legacy. You can adopt it for the philips

Nicolai- commented on blueprints/controllers/ikea_e1743/ikea_e1743.yaml:
```diff
@@ -249,24 +244,26 @@ variables:
+#**YAR** ADD - Logic to handle z2m legacy
+#**YAR** If user provided input for controller_entity assume legacy **Maybe There Is A Better Way To Do This**
+#**YAR** setting z2m_legacy to true with below code and set controller_id accordingly
+  z2m_legacy: '{{ integration_id == "zigbee2mqtt" and controller_entity not in [''] }}'
```
I get an error from this line when testing it.

Cause you escape the " jinja code" with single quotes, then you need to use double inside the jinja code or vice versa.

yarafie replied:
Very strange its the exact same code in the ikea_e2201 and is working for me.
Wait, I think it is because I'm not importing the blueprint but using it as is directly.
I'll check when I get home today.

So the line should be either:

z2m_legacy: "{{ integration_id == "zigbee2mqtt" and controller_entity not in [''] }}"
or
z2m_legacy: '{{ integration_id == "zigbee2mqtt" and controller_entity not in [""] }}'

EPMatt replied:
Thanks for spotting this @Nicolai-. Agree. Not sure whether it's needed, but I'd also check for `None` and `undefined`
```suggestion
  z2m_legacy: '{{ integration_id == "zigbee2mqtt" and controller_entity not in ["", None, undefined] }}'
```

yarafie replied:
Done. Will test when I get back home

Nicolai- commented on blueprints/controllers/ikea_e1743/ikea_e1743.yaml:
```diff
@@ -304,27 +301,27 @@ condition:
-      - '{{ integration_id != "zigbee2mqtt_v2" or trigger.payload != last_controller_event }}'
-      #YAR - ADD END
+#**YAR** - Below needed to change if z2m entity or device not sure I got this right
+#**YAR**      - '{{ integration_id != "zigbee2mqtt" or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
```
When looking at traces I don't think is right (same issue with my philips dimmer switch blueprint)

`last_controller_event` seems to not be declared in this scope, it never stops the execution here on traces with the same "action"

I don't believe it's an issue though for the functionality, as it has the `integrations_with_prev_event_storage` implemented, which makes sure duplicate events are not fired for the hooks.

Of course it would be nice/better/more performant to stop execution as early in the process as possible, but without loading the last action from the helper, I don't see how to accomplish that.

I would suggest simply setting the condition to
`integration_id != "zigbee2mqtt" or z2m_legacy == false or trigger.event.data.new_state.state != trigger.event.data.old_state.state` as we don't have the "old action" easy available.

@EPMatt any thoughts about this? Since you are the one with the knowledge about how it was designed.

yarafie replied:
Yep. I had a feeling it was wrong.

EPMatt replied:
@Nicolai- @yarafie some time passed since I wrote this, so I might be not completely correct. I think the goal of this check was specifically to prevent even starting the automation when the `controller_entity` overall state changes (e.g. due to an attribute change, like battery, link quality etc.) but the `.state` attribute change does not. In such situation, the state change is completely irrelevant to our automation and we need to filter it out.

Assuming the context in my old code comments is accurate and still relevant, see suggestion in my next comment:

Nicolai- reviewed (changes requested):
I also have a single E1743 remote, so went ahead and tested this.

yarafie committed:
fixed z2m_legacy= code

EPMatt commented on blueprints/controllers/ikea_e1743/ikea_e1743.yaml:
```diff
@@ -261,9 +315,13 @@ condition:
-      - '{{ integration_id != "zigbee2mqtt" or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
+#**YAR** - Below needed to change if z2m entity or device not sure I got this right
+#**YAR**      - '{{ integration_id != "zigbee2mqtt" or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
+      - '{{ integration_id != "zigbee2mqtt" or trigger.payload != last_controller_event or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
```
```suggestion
      - '{{ not (integration_id == "zigbee2mqtt" and z2m_legacy) or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
```

@Nicolai- equivalent to what you suggested, but that should make it clearer that the first two conditions are related.

Nicolai- replied:
Almost, I removed the `or trigger.payload != last_controller_event` part as this part was always true, when looking at my traces
I'm pretty sure its because `last_controller_event` is not defined in this scope, it's defined in the action scope, but from what I can read these variables are only available in that scope.

EPMatt replied:
EDIT: Correct! Not sure how that ended up there, thanks for spotting that. Edited my suggestion according to your comment.

EPMatt commented on blueprints/controllers/ikea_e1743/ikea_e1743.yaml:
```diff
@@ -261,9 +315,13 @@ condition:
         {%- endset -%}
         {{ trigger_action not in ["","None"] }}
+#**YAR** - ADD END
       # only for zigbee2mqtt, check if the event is relative to a real state change, and not only some minor changes in the sensor attributes
```
```suggestion
      # only for zigbee2mqtt in legacy mode, check if the event is relative to a real state change, and not only some minor changes in the sensor attributes
```

Update comment matching the new condition.

EPMatt commented:
> Next Steps?

@yarafie thanks for your contribution. 🚀  For what I see the current situation is, we're pretty close to have a fully working version for the IKEA E1743. 
If @Nicolai- is able to iterate on this with another round of testing, once we fixed my last code suggestions above, I believe we can open a PR to the original repo just with the E1743 changes. First, let's wait for confirmation that the E1743 controller blueprint from this PR is fully working - so we keep the review and fix process here instead of the new PR.

Then the easiest process for you for opening the new PR would be:
1. Create a new fork from `EPMatt/awesome-ha-blueprints`
2. In the new fork, create and switch to a new branch `ikea-e1743-device-triggers`
3. Copy the `ikea_e1743.yaml` file from `yarafie/awesome-ha-blueprints` `main` branch (which includes the changes from this PR)
4. Paste the file content on top of `ikea_e1743.yaml` in the `ikea-e1743-device-triggers` branch in the new fork
5. Open a PR on the original repo:  head: `new fork/ikea-e1743-device-triggers` -> base: `EPMatt/awesome-ha-blueprints/main`

If you agree, I'd say first let's make sure we have a working version for the E1743 if it does not take too much time. Then, we can take if from there with all the other controllers + the E2201. 👍🏻

yarafie committed:
Fixed check if event is relative to a real state change

yarafie committed:
Updated version

Nicolai- commented:
> > Next Steps?
> 
> @yarafie thanks for your contribution. 🚀 For what I see the current situation is, we're pretty close to have a fully working version for the IKEA E1743. If @Nicolai- is able to iterate on this with another round of testing, once we fixed my last code suggestions above, I believe we can open a PR to the original repo just with the E1743 changes. First, let's wait for confirmation that the E1743 controller blueprint from this PR is fully working - so we keep the review and fix process here instead of the new PR.
> 
> Then the easiest process for you for opening the new PR would be:
> 
> 1. Create a new fork from `EPMatt/awesome-ha-blueprints`
> 2. In the new fork, create and switch to a new branch `ikea-e1743-device-triggers`
> 3. Copy the `ikea_e1743.yaml` file from `yarafie/awesome-ha-blueprints` `main` branch (which includes the changes from this PR)
> 4. Paste the file content on top of `ikea_e1743.yaml` in the `ikea-e1743-device-triggers` branch in the new fork
> 5. Open a PR on the original repo:  head: `new fork/ikea-e1743-device-triggers` -> base: `EPMatt/awesome-ha-blueprints/main`
> 
> If you agree, I'd say first let's make sure we have a working version for the E1743 if it does not take too much time. Then, we can take if from there with all the other controllers + the E2201. 👍🏻

The IKEA E1743 works as expected here. No issues with the latest changes

yarafie commented:
> > Next Steps?
> 
> @yarafie thanks for your contribution. 🚀 For what I see the current situation is, we're pretty close to have a fully working version for the IKEA E1743. If @Nicolai- is able to iterate on this with another round of testing, once we fixed my last code suggestions above, I believe we can open a PR to the original repo just with the E1743 changes. First, let's wait for confirmation that the E1743 controller blueprint from this PR is fully working - so we keep the review and fix process here instead of the new PR.
> 
> Then the easiest process for you for opening the new PR would be:
> 
> 1. Create a new fork from `EPMatt/awesome-ha-blueprints`
> 2. In the new fork, create and switch to a new branch `ikea-e1743-device-triggers`
> 3. Copy the `ikea_e1743.yaml` file from `yarafie/awesome-ha-blueprints` `main` branch (which includes the changes from this PR)
> 4. Paste the file content on top of `ikea_e1743.yaml` in the `ikea-e1743-device-triggers` branch in the new fork
> 5. Open a PR on the original repo:  head: `new fork/ikea-e1743-device-triggers` -> base: `EPMatt/awesome-ha-blueprints/main`
> 
> If you agree, I'd say first let's make sure we have a working version for the E1743 if it does not take too much time. Then, we can take if from there with all the other controllers + the E2201. 👍🏻

Easier said than done :) since I already have a fork from lsismerio which is forked from EPMatt I can't make a new fork. I might have to delete the current fork first.
I'm not that well versed in github so if you have sny ideas most welcomed :)

Nicolai- commented:
> Easier said than done :) since I already have a fork from lsismerio which is forked from EPMatt I can't make a new fork. I might have to delete the current fork first.
> I'm not that well versed in github so if you have sny ideas most welcomed :)

 @yarafie 
I believe instead of step 1 and 2, then you should do this.
When creating a new branch you can chose the "source branch" and here you should be able to select the one from @EPMatt.

yarafie committed:
Minor Changes

yarafie commented:
> > Easier said than done :) since I already have a fork from lsismerio which is forked from EPMatt I can't make a new fork. I might have to delete the current fork first.
> > I'm not that well versed in github so if you have sny ideas most welcomed :)
> 
> @yarafie I believe instead of step 1 and 2, then you should do this. When creating a new branch you can chose the "source branch" and here you should be able to select the one from @EPMatt.

Nope. It only gives me option to create branch either in my own fork or from lsismerio
I may have to delete my current fork and re-fork from EPMatt.

Nicolai- commented:
> > > Easier said than done :) since I already have a fork from lsismerio which is forked from EPMatt I can't make a new fork. I might have to delete the current fork first.
> > > I'm not that well versed in github so if you have sny ideas most welcomed :)
> > 
> > 
> > @yarafie I believe instead of step 1 and 2, then you should do this. When creating a new branch you can chose the "source branch" and here you should be able to select the one from @EPMatt.
> 
> Nope. It only gives me option to create branch either in my own fork or from lsismerio I may have to delete my current fork and re-fork from EPMatt.

Can you make a new branch, copy the changes and then make a pull request against the original repo?

yarafie commented:
> > > > Easier said than done :) since I already have a fork from lsismerio which is forked from EPMatt I can't make a new fork. I might have to delete the current fork first.
> > > > I'm not that well versed in github so if you have sny ideas most welcomed :)
> > > 
> > > 
> > > @yarafie I believe instead of step 1 and 2, then you should do this. When creating a new branch you can chose the "source branch" and here you should be able to select the one from @EPMatt.
> > 
> > 
> > Nope. It only gives me option to create branch either in my own fork or from lsismerio I may have to delete my current fork and re-fork from EPMatt.
> 
> Can you make a new branch, copy the changes and then make a pull request against the original repo?

Google is my best Friend. PR done 

@EPMatt 
Ikea e1743 device triggers
#659 opened 1 minute ago by [yarafie]

theincogtion commented:
Wow great to see your efforts here :)
Don't forget E2002 Styrbar.
https://github.com/lsismeiro/awesome-ha-blueprints/tree/main/blueprints/controllers/ikea_e2001_e2002

yarafie commented:
> Wow great to see your efforts here :) Don't forget E2002 Styrbar. https://github.com/lsismeiro/awesome-ha-blueprints/tree/main/blueprints/controllers/ikea_e2001_e2002

One by one.

franbugallo commented on blueprints/controllers/ikea_e1812/ikea_e1812.yaml:
```diff
@@ -32,12 +29,24 @@ blueprint:
+          filter:
+          - integration: mqtt
+            manufacturer: IKEA
+            model: TRADFRI shortcut button (E1812)
```
hi @yarafie I think there is a typo in model: I think you should remove `(E1812)`

I'm testing your branch with my E1812 device and I see no devices popping up. If I check in my devices: 

<img width="1026" alt="Captura de Pantalla 2025-01-07 a las 22 03 02" src="https://github.com/user-attachments/assets/425fd05a-3b16-4cd9-a4f2-468651cd6c9c" />
<img width="671" alt="Captura de Pantalla 2025-01-07 a las 22 05 40" src="https://github.com/user-attachments/assets/ee6a66ef-1c51-4a6f-a26f-673becf3f405" />

yarafie replied:
Thankyou for testing.
Not a typo.
Z2M 2.x changed the naming of devices they removed the model number from the model name and added it as a seperate model_id attribute.

I'm working on fixing all but in the main EPMatt branch,

yarafie committed:
Updated all with new zigbee2mqtt logic preserving legacy
This may break existing automations

yarafie committed:
Starting all comments in code from column 2
Prettier - Enforce code style

yarafie committed:
Adding space after # for comments
Prettier - Enforce code style

yarafie committed:
Prettier - Enforce code style

yarafie commented:
To all whom are using my fork.
In order to start fresh I'm going to delete my fork of lsismerio and refork from Original EPMatt in order for me to be able to create directly with the main Repo.
I will do this in about 2 hours from this comment.

EPMatt commented:
@yarafie @Nicolai- thanks for sorting out the issues with the already open fork, and for opening the PR on the original repo! :)

For anyone following this thread: Initial PR for MQTT Device triggers for the IKEA E1743 has been opened. 🔥 
https://github.com/EPMatt/awesome-ha-blueprints/pull/661

As soon as we merge that, we are gonna follow up with PRs for all the other controllers.
Thanks for everyone who contributed to the discussion here! Stay tuned and feel free to jump in the linked PR if you want to test the blueprints on your system and share your feedback.

Xander-V commented:
Great work guys, highly appreciated!
And also glad to see you back @EPMatt

If testing is needed, I will gladly help with:
* [E1524/E1810](https://www.zigbee2mqtt.io/devices/E1524_E1810.html#ikea-e1524%252Fe1810)
* [E2001/E2002](https://www.zigbee2mqtt.io/devices/E2001_E2002.html#ikea-e2001%252Fe2002)
